### PR TITLE
[NFC] `getopts` is used by `librustc` and `librustc_driver`, but isn't declared in their `Cargo.toml`.

### PR DESCRIPTION
**Pull request by DiamondLovesYou**

### CODE REVIEW

**Review by davidtwco — COMMENTED** [review]
Thanks for the PR! You'll need to update the `Cargo.lock` files in order for Travis to build and pass with this.

**Comment by DiamondLovesYou** [comment]
@davidtwco Done!

**Comment by kennytm** [comment]
@bors r-

Failed to build in `dist-various-1` (log: https://github.com/rust-lang/rust/pull/54790#issuecomment-426713905).

```
[00:07:42]    Compiling syntax_ext v0.0.0 (/checkout/src/libsyntax_ext)
[00:07:42] error[E0523]: found two different crates with name `getopts` that are not distinguished by differing `-C metadata`. This will result in symbol conflicts between the two.
[00:07:42]     |
[00:07:42] 119 | extern crate test;
[00:07:42]     | ^^^^^^^^^^^^^^^^^^
[00:07:42] 
...
[00:07:58] error: build failed
[00:07:58] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "build" "--target" "x86_64-unknown-linux-gnu" "-j" "4" "--release" "--locked" "--color" "always" "--features" " jemalloc" "--manifest-path" "/checkout/src/rustc/Cargo.toml" "--message-format" "json"
[00:07:58] expected success, got: exit code: 101
[00:07:58] thread 'main' panicked at 'cargo must succeed', bootstrap/compile.rs:1112:9
[00:07:58] travis_fold:end:stage0-rustc
```

**Comment by kennytm** [comment]
@bors try 

Verifying if this is caused by other PRs in the rollup or this PR itself.

**Comment by kennytm** [comment]
Oh well 🤷

File: src/librustc_driver/Cargo.toml [hunk 1]
```diff
@@ -15,6 +15,7 @@ log = "0.4"
 env_logger = { version = "0.5", default-features = false }
 rustc-rayon = "0.1.1"
 scoped-tls = { version = "0.1.1", features = ["nightly"] }
+getopts = "=0.2.17"
```
Comment (Mark-Simulacrum) [inline src/librustc_driver/Cargo.toml]: Why are these `=` restrictions?

**Review by Mark-Simulacrum — CHANGES_REQUESTED** [review]

File: src/librustc_driver/Cargo.toml (hunk 1 above)
Comment (DiamondLovesYou) [reply src/librustc_driver/Cargo.toml]: To keep the existing crate version, which is what is used by some other tool (not sure which). Shall I change it?

**Comment by TimNN** [comment]
Ping from triage @DiamondLovesYou: It looks like some changes are required for this PR.

**Comment by TimNN** [comment]
Ping from triage @DiamondLovesYou: What is the status of this PR?

**Pull request closed without merging**